### PR TITLE
Refactor issue filing to enable multiple result data types

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This PR follows up from #1351 . The current issue filing implementation assumes the issue-details type contains axe-core result data. This needs to change if we want to reuse the plumbing for the upcoming platform-agnostic data.
- `CreateIssueDetailsTextData` only holds the fields issue filing needs to create a bug, no more or less
- Each consumer of issue filing will convert their data-type (axe-core data or unified-data) to `CreateIssueDetailsTextData` (the cards UI will soon need to do this with unified-data)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/issue-filing/common/create-issue-details-builder.ts [hunk 1]
```diff
@@ -9,34 +9,33 @@ import { MarkupFormatter } from './markup/markup-formatter';
...
             sectionSeparator(),
 
             sectionHeader('Target application'),
             sectionHeaderSeparator(),
-            link(data.pageUrl, data.pageTitle),
+            link(data.targetApp.url, data.targetApp.name),
             sectionSeparator(),
 
-            sectionHeader('Element path'),
+            sectionHeader('Element'),
```
Comment (karanbirsingh) [inline src/issue-filing/common/create-issue-details-builder.ts]: This is the only part of this PR that affects the UI - Fer suggested it as a platform-agnostic label. May move into separate PR

File: src/issue-filing/common/issue-filing-url-string-utils.ts [hunk 2]
```diff
@@ -15,9 +15,7 @@ const getTitle = (data: CreateIssueDetailsTextData): string => {
         prefix = prefix + ': ';
     }
 
-    const selectorLastPart = getSelectorLastPart(data.ruleResult.selector);
-
-    return `${prefix}${data.ruleResult.help} (${selectorLastPart})`;
+    return `${prefix}${data.rule.description} (${data.element.conciseName})`;
 };
 
 const getSelectorLastPart = (selector: string): string => {
```
Comment (karanbirsingh) [inline src/issue-filing/common/issue-filing-url-string-utils.ts]: In a different PR we may want to move `getSelectorLastPart` out of this file because the other functions are platform-agnostic whereas this is web-specific.

File: src/issue-filing/rule-result-to-issue-filing-data.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (haonliu) [inline src/issue-filing/rule-result-to-issue-filing-data.ts]: There seems no unit test for this class?

Comment (karanbirsingh) [reply src/issue-filing/rule-result-to-issue-filing-data.ts]: Ah, thanks! mixed up my branches - let me port it over

File: src/tests/unit/tests/DetailsView/components/issues-details-pane.test.tsx [hunk 4]
```diff
@@ -111,6 +112,9 @@ describe('IssuesDetailsPaneTest', () => {
         }
 
         const deps: IssuesDetailsPaneDeps = {
+            axeResultToIssueFilingDataConverter: {
```
Comment (haonliu) [inline src/tests/unit/tests/DetailsView/components/issues-details-pane.test.tsx]: It would be better to mock this converter to return something meaningful. right now it renders undefined in the snapshot.

**Comment by haonliu** [comment]
LGTM in general, left a few minor comments.

File: src/tests/unit/tests/background/issue-details-text-generator.test.ts [hunk 5]
```diff
@@ -20,19 +20,23 @@ describe('Issue details text builder', () => {
...
-                help: 'RR-help',
-                html: 'RR-html',
-                ruleId: 'RR-rule-id',
-                helpUrl: 'RR-help-url',
-                selector,
-                snippet: 'RR-snippet   space',
-            } as any,
-        };
+            rule: {
+                description: 'RR-help',
```
Comment (lisli1) [inline src/tests/unit/tests/background/issue-details-text-generator.test.ts]: The values were probably kept the same to avoid updating the snapshot, but should we change the values to more closely match the new structure of the data? It might be confusing when looking at this test in the future why it says "RR" when it's not mapped to a `ruleResult` or "failureSummary" when the key is a `howToFixSummary`.

Comment (karanbirsingh) [reply src/tests/unit/tests/background/issue-details-text-generator.test.ts]: Good point! I think we should do it in a different PR. If the snapshot doesn't change in this PR it helps show that the behavior has not changed?

File: src/tests/unit/tests/common/message-creators/issue-filing-action-message-creator.test.ts [hunk 6]
```diff
@@ -77,36 +76,23 @@ describe('IssueFilingActionMessageCreator', () => {
...
-                    messageType: Messages.IssueFiling.FileIssue,
-                    payload: {
-                        service: testService,
-                        issueData: issueDetailsData,
-                        telemetry,
-                    },
-                }),
-            ),
-        );
+        const issueDetailsData: CreateIssueDetailsTextData = {} as any;
```
Comment (lisli1) [inline src/tests/unit/tests/common/message-creators/issue-filing-action-message-creator.test.ts]: Just wondering, why are we casting `issueDetailsData as any` here but using `issueDetailsData: {} as CreateIssueDetailsTextData` in issue-filing-button.test?

Comment (karanbirsingh) [reply src/tests/unit/tests/common/message-creators/issue-filing-action-message-creator.test.ts]: I think both are similar & don't know of any meaningful differences - afaik it's preference / what nearby code has already that might determine what one chooses
